extensions/amp-next-page/1.0/visibility-observer.js

```diff
@@ -178,4 +298,90 @@ export default class VisibilityObserver {
     installPositionObserverServiceForDoc(this.ampdoc_);
     return Services.positionObserverForDoc(this.ampdoc_.getHeadNode());
   }
+
+  /**
+   * Calculates the position of the element relative to the viewport
+   * based on the positions of the injected bottom and top sentinel elements
+   *
+   * @param {!VisibilityObserverEntry} entry
+   */
+  updateRelativePos(entry) {
+    const relativePos = entry.usesSentinel()
+      ? this.getRelativePosFromSentinel(entry)
+      : this.getRelativePosFromScroll(
+          /** @type {!VisibilityObserverHostEntry} */ (entry)
+        );
+    if (!relativePos) {
+      return;
+    }
+    entry.callback(devAssert(relativePos));
+  }
+
+  /**
+   * @param {!VisibilityObserverHostEntry} entry
+   * @return {?ViewportRelativePos}
+   */
+  getRelativePosFromScroll(entry) {
+    // Measure the position of the host page (edge case)
+    const {viewportHeight_: vh, lastScrollTop_: scroll} = this;
+    // Document height is the same as the distance from the top
+    // to the <amp-next-page> element
+    const {top: height} = entry.nextPageEl.getLayoutBox();
```

Comment:
This kind of sync access to scroll positions will be deprecated soon.

Reply:
Added a `TODO`